gui/src/main/window_behaviour.js

```diff
@@ -1,29 +1,104 @@
-const {shell} = require('electron')
+const {BrowserWindow, ipcMain, shell} = require('electron')
+const path = require('path')
 
-module.exports.openExternalLinks = (win) => {
-  win.webContents.on('will-navigate', (event, url) => {
-    if (url.startsWith('http') && !url.match('/auth/authorize')) {
-      event.preventDefault()
-      shell.openExternal(url)
-    }
-  })
-  return win
-}
+const ELMSTARTUP = 400
 
-module.exports.noMenu = (win) => {
-  win.setMenu(null)
-  win.setAutoHideMenuBar(true)
-}
+const log = require('cozy-desktop').default.logger({
+  component: 'windows'
+})
+
+module.exports = class WindowManager {
```

Comment:
Maybe rename `gui/src/main/window{_behaviour -> _manager}.js`?
Or even remove suffix and rename `WindowManager` to `Window`?
(can be done later)

I wonder whether we could have used composition over inheritance (we're just factoring code here, there's no need to provide multiple `WindowManager` implementations to some consumer code)... But it wouldn't be so easy to achieve, event handlers would be particularly hard to extract...

And the new version is definitely a huge improvement anyway 👍